get site group if it exists

## Detailed Description
Currrently when making a new user, if it doesnt exsits, a new user and site group is made.
Sometimes the site group already exists, so we should laod that, not cause an error by overwriting it

## Context
- good to reduce errors

## Possible Implementation
in the code [here](https://github.com/openclimatefix/pv-site-datamodel/blob/main/pvsite_datamodel/read/user.py#L29), try to get a site group with this email first, and if it doesnt exsits, then make it. 
it would also be nice to use [create_site_group](https://github.com/openclimatefix/pv-site-datamodel/blob/main/pvsite_datamodel/write/user_and_site.py#L47) function to reduce code

FAILED tests/test_alembic.py::test_migrations - docker.errors.DockerException: Error while fetching server API version: (2, 'CreateFile', 'The system cannot find the file specified.')

Is this because I haven't started a local database in docker?

--- a/pvsite_datamodel/read/user.py
+++ b/pvsite_datamodel/read/user.py
@@ -7,7 +7,6 @@ from typing import List, Optional
 from sqlalchemy.orm import Session, contains_eager
 
 from pvsite_datamodel.sqlmodels import APIRequestSQL, SiteGroupSQL, UserSQL
-
 logger = logging.getLogger(__name__)
 
 
@@ -26,10 +25,15 @@ def get_user_by_email(session: Session, email: str, make_new_user_if_none: bool 
     if user is None and make_new_user_if_none is True:
         logger.info(f"User with email {email} not found, so making one")
 
-        # making a new site group
-        site_group = SiteGroupSQL(site_group_name=f"site_group_for_{email}")
-        session.add(site_group)
-        session.commit()
+        #checking for site_group
+        site_group = (
+        session.query(SiteGroupSQL).filter(SiteGroupSQL.site_group_name == f"site_group_for_{email}").first()
+    )
+        # making a new site group if one doesn't exist
+        if site_group is None:
+            site_group = SiteGroupSQL(site_group_name=f"site_group_for_{email}")
+            session.add(site_group)
+            session.commit()
 
         # make a new user
         user = UserSQL(email=email, site_group_uuid=site_group.site_group_uuid)

--- a/pvsite_datamodel/read/user.py
+++ b/pvsite_datamodel/read/user.py
@@ -7,6 +7,7 @@ from typing import List, Optional
 from sqlalchemy.orm import Session, contains_eager
 
 from pvsite_datamodel.sqlmodels import APIRequestSQL, SiteGroupSQL, UserSQL
+
 logger = logging.getLogger(__name__)
 
 
@@ -25,10 +26,12 @@ def get_user_by_email(session: Session, email: str, make_new_user_if_none: bool 
     if user is None and make_new_user_if_none is True:
         logger.info(f"User with email {email} not found, so making one")
 
-        #checking for site_group
+        # checking for site_group
         site_group = (
-        session.query(SiteGroupSQL).filter(SiteGroupSQL.site_group_name == f"site_group_for_{email}").first()
-    )
+            session.query(SiteGroupSQL)
+            .filter(SiteGroupSQL.site_group_name == f"site_group_for_{email}")
+            .first()
+        )
         # making a new site group if one doesn't exist
         if site_group is None:
             site_group = SiteGroupSQL(site_group_name=f"site_group_for_{email}")

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,3 +1,4 @@
+
 [tool.poetry]
 name = "pvsite-datamodel"
 version = "1.0.36"

--- a/tests/read/test_get_user_by_email.py
+++ b/tests/read/test_get_user_by_email.py
@@ -1,11 +1,15 @@
-from pvsite_datamodel import UserSQL
+from pvsite_datamodel import UserSQL, SiteGroupSQL
 from pvsite_datamodel.read import get_user_by_email
 from pvsite_datamodel.write.user_and_site import create_site_group, create_user
 
 
 class TestGetUserByEmail:
     """Test for get_user_by_email function"""
-
+    def test_get_user_by_email_no_users_site_group_exists(self,db_session):
+        site_group = create_site_group(db_session=db_session,site_group_name="site_group_for_test@test.com")
+        user = get_user_by_email(session=db_session, email="test@test.com")
+        assert len(db_session.query(UserSQL).all()) == 1
+        assert len(db_session.query(SiteGroupSQL).all()) == 1
     def test_get_user_by_email_no_users(self, db_session):
         user = get_user_by_email(session=db_session, email="test@test.com")
         assert user.email == "test@test.com"

--- a/tests/read/test_get_user_by_email.py
+++ b/tests/read/test_get_user_by_email.py
@@ -1,15 +1,17 @@
-from pvsite_datamodel import UserSQL, SiteGroupSQL
+from pvsite_datamodel import SiteGroupSQL, UserSQL
 from pvsite_datamodel.read import get_user_by_email
 from pvsite_datamodel.write.user_and_site import create_site_group, create_user
 
 
 class TestGetUserByEmail:
     """Test for get_user_by_email function"""
-    def test_get_user_by_email_no_users_site_group_exists(self,db_session):
-        site_group = create_site_group(db_session=db_session,site_group_name="site_group_for_test@test.com")
-        user = get_user_by_email(session=db_session, email="test@test.com")
+
+    def test_get_user_by_email_no_users_site_group_exists(self, db_session):
+        _ = create_site_group(db_session=db_session, site_group_name="site_group_for_test@test.com")
+        _ = get_user_by_email(session=db_session, email="test@test.com")
         assert len(db_session.query(UserSQL).all()) == 1
         assert len(db_session.query(SiteGroupSQL).all()) == 1
+
     def test_get_user_by_email_no_users(self, db_session):
         user = get_user_by_email(session=db_session, email="test@test.com")
         assert user.email == "test@test.com"
@@ -17,10 +19,10 @@ class TestGetUserByEmail:
 
     def test_get_user_by_email_with_users(self, db_session):
         site_group = create_site_group(db_session=db_session)
-        user = create_user(
+        _ = create_user(
             session=db_session, site_group_name=site_group.site_group_name, email="test_1@test.com"
         )
-        user = create_user(
+        _ = create_user(
             session=db_session, site_group_name=site_group.site_group_name, email="test_2@test.com"
         )
 

